pkg/helm/actions/setup_test.go

```diff
@@ -0,0 +1,55 @@
+package actions
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+
+	"helm.sh/helm/v3/pkg/cli"
+)
+
+func setSettings(settings *cli.EnvSettings) {
+	settings.RepositoryCache = os.TempDir()
+	settings.RegistryConfig = os.TempDir()
+	settings.RepositoryConfig = "/RepositoryConfig"
+}
+
+func TestMain(m *testing.M) {
+	setupTestWithTls()
+	setupTestWithoutTls()
+	retCode := m.Run()
+	ExecuteScript("./testdata/cleanupNonTls.sh")
+	ExecuteScript("./testdata/cleanup.sh")
+	os.Exit(retCode)
+}
+
+func setupTestWithTls() {
+	setSettings(settings)
+	ExecuteScript("./testdata/createTlsSecrets.sh")
+	//start chartmuseum server
+	ExecuteScript("./testdata/chartmuseum.sh")
+	ExecuteScript("./testdata/cacertCreate.sh")
+	ExecuteScript("./testdata/uploadCharts.sh")
+}
+
+func setupTestWithoutTls() {
+	setSettings(settings)
+	ExecuteScript("./testdata/chartmuseumWithoutTls.sh")
+	ExecuteScript("./testdata/uploadChartsWithoutTls.sh")
+}
+
+func ExecuteScript(filepath string) error {
+	tlsCmd := exec.Command(filepath)
+	tlsCmd.Stdout = os.Stdout
+	err := tlsCmd.Start()
+	if err != nil {
+		return err
+	}
+	if filepath != "./testdata/chartmuseum.sh" && filepath != "./testdata/chartmuseumWithoutTls.sh" {
```

Comment:
Make this boolean an argument, and decide whether wait should be called on the caller.

Reply:
Yeah makes sense